source_files/coal/c_compile.cc

```diff
@@ -581,21 +581,21 @@ Definition *RealVM::NewGlobal(Type *type)
     int tsize = type_size[type->type];
 
     Definition *var = new Definition;
-    memset(var, 0, sizeof(Definition));
+    EPI_CLEAR_MEMORY(var, Definition, 1);
 
     var->ofs  = global_mem_.Alloc(tsize * sizeof(double));
     var->type = type;
 
     // clear it
-    memset(global_mem_.Deref(var->ofs), 0, tsize * sizeof(double));
+    EPI_CLEAR_MEMORY((double *)global_mem_.Deref(var->ofs), double, tsize);
```

Comment:
It would be a good idea to run an ASAN pass after landing this, to catch anywhere maybe don't have the right type/size specified

Reply:
Ran some sessions with ASAN enabled, oddly enough just caught several issues unrelated to this